pkg/kapp/resources/resources.go

```diff
@@ -472,13 +475,24 @@ func (c *ResourcesImpl) resourceErr(err error, action string, resource Resource)
 	return resourcePlainErr{err, action, resource}
 }
 
-func (c *ResourcesImpl) resourceClient(resource Resource) (dynamic.ResourceInterface, ResourceType, error) {
+type resourceClientOpts struct {
+	Warnings bool
+}
+
+func (c *ResourcesImpl) resourceClient(resource Resource, opts resourceClientOpts) (dynamic.ResourceInterface, ResourceType, error) {
```

Comment:
Should we make the options optional?

Reply:
All the usages are within this file so updating them wasn't a hassle. Doesn't a struct lead to a more robust function signature that way?

Reply:
It's just that the struct is not required except for one place and we are kind of forcing it's usage everywhere.

Reply:
That is fair. I am still torn about using the triple dot operator to mark an optional parameter as it's actual intention is to accept a slice of values

Reply:
... for optional params is really handy for public APIs to preserve backwards compat. in this case its all internal so i would not recommend going with ....

Reply:
Noted! Resolving this thread for now